File: dotnet/src/xjustiz.core-dotnet/Models/Entities/Core/AkteCore.cs
namespace xjustiz.core_dotnet.Models.Entities.Core;

using System.Xml.Serialization;
using xjustiz.core_dotnet.Models.Helpers;
using xjustiz.core_dotnet.Util.Versioning;

/// <summary>
/// Repräsentiert eine Akte.<br/>
/// <u><b>File:</b></u> Represents a file / case.
/// </summary>
[XJustizCoreAvailability(XJustizCoreVersion.V1_0_0)]
public class AkteCore : Akte
{
    /// <summary>
    /// Fachspezifische Daten einer Akte.<br/>
    /// <u><b>Subject-specific data:</b></u> Specialized data of a file/case.
    /// </summary>
    [XmlElement("xjustiz.fachspezifischeDaten", Namespace = XJustizConstants.Tns, Order = 3)]
    [XJustizAvailability(XJustizVersion.V3_1_1)]
    [XJustizCoreAvailability(XJustizCoreVersion.V1_0_0)]
    public new XjustizAkteFachspezifischeDatenCore FachspezifischeDaten { get; set; } = new();
}

File: sdk/java/src/main/java/de/xjustiz/core/models/AkteCore.java
package de.xjustiz.core.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.xml.bind.annotation.XmlAccessType;
import jakarta.xml.bind.annotation.XmlAccessorType;
import jakarta.xml.bind.annotation.XmlType;

/**
 * Repräsentiert eine Akte.
 * <p>
 * <u><b>File:</b></u> Represents a file / case.
 */
@XmlType
@XmlAccessorType(XmlAccessType.FIELD)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class AkteCore extends Akte {

    /**
     * Default constructor.
     */
    public AkteCore() {}

}